### ISSUE TITLE
Issue with creating a FileStorage with Terraform

### ISSUE THREAD

**Issue by imanzhulin** [IBM-Cloud/terraform-provider-ibm#2251]
### Terraform Version
0.13.4

### Affected Resource(s)
Please list the resources as a list, for example:
- ibm_storage_file

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
data "ibm_resource_group" "vibium_group" {
  name = var.vibium_resource_group
}

data "ibm_is_image" "vibium_ubuntu" {
  name = var.image_name
}

data "ibm_is_ssh_key" "vibium_ssh_key" {
  name = var.ssh_key
}

resource "ibm_is_vpc" "vibium_vpc" {
  name = "${var.vibium_resource_group}-vpc"
  resource_group = data.ibm_resource_group.vibium_group.id
}

resource "ibm_is_instance" "vibium_api" {
  name    = "${var.resources_prefix}-${var.vibium_resource_group}-api"
  vpc     = ibm_is_vpc.vibium_vpc.id
  zone    = "${var.vpc_region}-1"
  keys    = [data.ibm_is_ssh_key.vibium_ssh_key.id]
  image   = data.ibm_is_image.vibium_ubuntu.id
  profile = var.vibium_vpc_api_image_profile
  resource_group = data.ibm_resource_group.vibium_group.id

  primary_network_interface {
    subnet          = ibm_is_subnet.vibium_subnet.id
    security_groups = [ibm_is_security_group.vibium_sg.id]
  }
}

resource "ibm_is_floating_ip" "vibium_api_fip" {
  name   = "${var.vibium_resource_group}-api-fip"
  target = ibm_is_instance.vibium_api.primary_network_interface[0].id
  resource_group = data.ibm_resource_group.vibium_group.id
}

resource "ibm_storage_file" "vibium_api_filestorage" {
  type       = "Endurance"
  datacenter = "fra04"
  capacity   = 4000
  iops       = 10

  # Optional fields
  #allowed_ip_addresses      = [ibm_is_floating_ip.vibium_api_fip.address]
  snapshot_capacity         = 4000
  hourly_billing            = true
  tags                      = ["${var.vibium_resource_group}-api","nfs"] 

  # Optional fields for snapshot
  snapshot_schedule {
    schedule_type   = "DAILY"
    retention_count = 10
    minute          = 30
    hour            = 2
    enable          = true
  }
}
```

### Debug Output
https://gist.github.com/imanzhulin/f16771af6ddf51e19531bc2b5aab252b

### Expected Behavior
File storage should be provisioned using Terraform.

### Actual Behavior
Terraform faces error:
**Error: Error while creating storage:SoftLayer_Exception_Public: Access Denied.  (HTTP 401)

  on infra_vibium.tf line 39, in resource "ibm_storage_file" "vibium_api_filestorage":
  39: resource "ibm_storage_file" "vibium_api_filestorage" {
Error: Process completed with exit code 1.**

### Steps to Reproduce
terraform plan -var='ibmcloud_api_key=${{ secrets.IBM_API_KEY }}' \
        -var='iaas_classic_username=${{ secrets.IAAS_CLASSIC_USERNAME }}' \
        -var='iaas_classic_api_key=${{ secrets.IAAS_CLASSIC_API_KEY }}' \
        -var-file='variables.tfvars' -out tf.out
        terraform apply -auto-approve tf.out

But I can create File Storage using UI (manually). The Classic Infra API Key was reset. It did not help.

**Comment by hkantare** [comment]
Can you please validate your credentials
Here your IAAS Classic User name refers to the VPN username...(API username)
```
To create or find your iaas_classic_username with the VPN credentials, refer to VPN credentials.
```
In the IBM Cloud® console, click Manage > Access (IAM), and select Users.
Select a user from the list.
From the User details page, go to the VPN section and get your VPN username
```

**Comment by imanzhulin** [comment]
![image](https://user-images.githubusercontent.com/64901273/107632670-d9728080-6c77-11eb-99f3-9803bdff264c.png)
I guess 2146184 [ ivan.manzhulin@vikinganalytics.se ]

**Comment by imanzhulin** [comment]
And one more thing if you need it. Current API keys I am using:
![image](https://user-images.githubusercontent.com/64901273/107632851-1c345880-6c78-11eb-86ad-50900f7876b9.png)

**Comment by hkantare** [comment]
The username is 2146184_ivan.manzhulin@vikinganalytics.se
can you try this sample API just to check credentials are proper

https://api.softlayer.com/rest/v3/SoftLayer_Product_Package/getAllObjects.json?objectFilter=%7B%22type%22%3A%7B%22keyName%22%3A%7B%22operation%22%3A%22STORAGE_AS_A_SERVICE%22%7D%7D%7D&objectMask=mask%5Bid%2Cname%2Cdescription%2CisActive%2Ctype%5BkeyName%5D%5D&resultLimit=0%2C1

With the credentials `2146184_ivan.manzhulin@vikinganalytics.se` and `<classicInfrastructureAPikey`

**Comment by imanzhulin** [comment]
I guess it worked.

**{"id":759,"isActive":1,"name":"Storage As A Service (StaaS)","type":{"keyName":"STORAGE_AS_A_SERVICE"}}**

**Comment by imanzhulin** [comment]
It seems to me that I should update IAAS_CLASSIC_USERNAME. I am using GithubActions secrets for storing sensitive info for Terraform. Please let me try

**Comment by imanzhulin** [comment]
No, unfortunately it did not help. The same error

**Comment by hkantare** [comment]
It should work can you enable the trace export TF_LOG=debug
Can you check once again the username and apikey which you are getting from Githubactions..

**Comment by hkantare** [comment]
I have one concern why are you using the classic infrastructure file storage for VPC infrastructure...
I don't think you can associate a classic infrastructure file storage to a VPC infrastructure...


For VPC infrastructure currently it supports only block volume...
https://cloud.ibm.com/docs/ibm-cloud-provider-for-terraform?topic=ibm-cloud-provider-for-terraform-vpc-gen2-resources#volume

**Comment by imanzhulin** [comment]
Do you have plans to add FS to VPN infra later? A similar request was raised when I tried to provision GPU instances with Terraform. So I guess for me it's easier to create all manually.

**Comment by imanzhulin** [comment]
And why in this case I need variables for Classic Infrastructure in Terraform if I can't use them?

**Comment by hkantare** [comment]
Can you please enable the trace by exporting environmental variable
export TF_LOG=debug
and share us the complete log of terraform apply just to confirm why IAAS credentials not working.

The IASSS credential are required only to provision the classic Infrastructure resources 

https://cloud.ibm.com/docs/ibm-cloud-provider-for-terraform?topic=ibm-cloud-provider-for-terraform-infrastructure-resources

Adding FS to VPN infra is a request needs to be raised on VPC infrastructure ...


We are not clear with GPU instance provsioning ..Are you provsioning the GPU instance in VPC or classic Infrastructure.

Can you let me know the brief manual steps how is GPU instance provsioined?

**Comment by imanzhulin** [comment]
Let close the issue as I understood that we can't mount File Storage to VPC Infra Virtual Server. I decided to create a block storage instead which also can be backed up and costs a bit cheaper than FS. 
And I also could not incorporate GPU instance in VPC infra using terraform since GPU instance belongs to Classic Infrastructure. I think these issues occur due to lack of documentation. I've already created a request for that.
Anyway thanks for the help.
